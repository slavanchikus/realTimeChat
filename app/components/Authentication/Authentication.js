import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './Authentication.module.styl';

export default class Authentication extends Component {
  static propTypes = {
    onUserRequest: PropTypes.func.isRequired,
    onUserCreate: PropTypes.func.isRequired,
  };

  state = {
    username: '',
    password: '',
    isRegistration: false
  };

  handleClick = () => {
    const { username, password, isRegistration } = this.state;
    if (isRegistration) {
      this.props.onUserCreate(username, password);
    } else {
      this.props.onUserRequest(username, password);
    }
  };

  putUsername = (e) => {
    this.setState({ username: e.target.value });
  };

  putPassword = (e) => {
    this.setState({ password: e.target.value });
  };

  render() {
    const { isRegistration } = this.state;
    return (
      <div className={styles.container}>
        <h2>
          {isRegistration ? 'Регистрация' : 'Авторизация'}
        </h2>
        <input type="text" placeholder="Юзернейм" onChange={this.putUsername} />
        <input type="text" placeholder="Пароль" onChange={this.putPassword} />
        <div className={styles.button} onClick={this.handleClick}>
          {isRegistration ? 'Регистрировать' : 'Войти'}
        </div>
        <div className={styles.choose} onClick={() => this.setState({ isRegistration: !this.state.isRegistration })}>
          {isRegistration ? 'Авторизоваться' : 'Зарегистрироваться'}
        </div>
      </div>
    );
  }
}
