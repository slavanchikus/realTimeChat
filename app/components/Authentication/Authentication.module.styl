.container
  height: 200px
  width: 300px
  padding: 35px 0
  background: #181818
  display: flex
  align-items: center
  flex-direction: column
  justify-content: space-around
  font-size: 13px
  input
    height: 22px
    padding: 4px
    border: 0
    outline: 0
    border-radius: 2px
  h2
    margin: 0

.button
  width: 177px
  height: 30px
  text-align: center
  line-height: 30px
  background: #ffdb4d
  border-color: rgba(191,153,0,.8)
  color: black
  cursor: pointer
  border-radius: 2px
  transition-duration: .2s;
  &:hover
    background: #ECC948

.choose
  margin-top: 10px
  cursor: pointer
  &:hover
    color: #FF6557
