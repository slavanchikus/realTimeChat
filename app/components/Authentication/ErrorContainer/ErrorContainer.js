import React from 'react';

import styles from './ErrorContainer.module.styl';

export default ({ error }) => (
  <div className={styles.container}>
    {error === 'invalid data' && 'Неправильный логин или пароль'}
    {error === 'user exists' && 'Юзернейм занят'}
  </div>
  );
