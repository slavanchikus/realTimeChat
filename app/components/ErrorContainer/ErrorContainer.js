import React from 'react';

import styles from './ErrorContainer.module.styl';

export default ({ error }) => (
  <div className={styles.container}>
    {error === 'invalid user data' && 'Неправильный логин или пароль'}
    {error === 'user exists' && 'Юзернейм занят'}
  </div>
  );
