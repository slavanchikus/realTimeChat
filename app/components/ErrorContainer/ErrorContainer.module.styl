.container
  position: absolute
  bottom: 15px
  font-size: 13px
  color: #ffdb4d;