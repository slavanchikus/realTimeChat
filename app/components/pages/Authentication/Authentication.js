import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import ErrorContainer from '../../shared/ErrorContainer/ErrorContainer';

import styles from './Authentication.module.styl';

export default class Authentication extends PureComponent {
  static propTypes = {
    errors: PropTypes.object.isRequired,
    onUserRequest: PropTypes.func.isRequired,
    onUserCreate: PropTypes.func.isRequired,
  };

  state = {
    username: '',
    password: '',
    isRegistration: false
  };

  handleClick = () => {
    const { username, password, isRegistration } = this.state;
    const usernameLen = !/^\s+$/.test(username) ? username.length : 0;
    const passwordLen = !/^\s+$/.test(password) ? password.length : 0;
    if (usernameLen > 1 && passwordLen > 1) {
      if (isRegistration) {
        this.props.onUserCreate(username, password);
      } else {
        this.props.onUserRequest(username, password);
      }
    }
  };

  putUsername = (e) => {
    this.setState({ username: e.target.value });
  };

  putPassword = (e) => {
    this.setState({ password: e.target.value });
  };

  render() {
    const { errors } = this.props;
    const { isRegistration, username, password } = this.state;
    return (
      <div className={styles.container}>
        <h2>
          {isRegistration ? 'Регистрация' : 'Авторизация'}
        </h2>
        <input type="text" placeholder="Юзернейм" onChange={this.putUsername} value={username} maxLength={15} />
        <input type="password" placeholder="Пароль" onChange={this.putPassword} value={password} maxLength={10} />
        <div className={styles.button} onClick={this.handleClick}>
          {isRegistration ? 'Регистрировать' : 'Войти'}
        </div>
        <div className={styles.choose} onClick={() => this.setState({ isRegistration: !this.state.isRegistration, username: '', password: '' })}>
          {isRegistration ? 'Авторизоваться' : 'Зарегистрироваться'}
        </div>
        {errors.userError &&
          <ErrorContainer
            error={errors.userError}
          />}
      </div>
    );
  }
}
