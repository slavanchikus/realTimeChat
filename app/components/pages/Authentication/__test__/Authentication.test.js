import React from 'react';
import { mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import Authentication from '../Authentication';
import ErrorContainer from '../../../shared/ErrorContainer/ErrorContainer';

configure({ adapter: new Adapter() });

describe('Authentication', () => {
  let props;
  const authenticationWrapper = newProps => (
    <Authentication {...newProps} />
  );

  describe('when requred props are defined for reply Authentication', () => {
    beforeEach(() => {
      props = {
        errors: {},
        onUserRequest: jest.fn(),
        onUserCreate: jest.fn()
      };
    });

    it('renders correctly', () => {
      const wrapper = mount(authenticationWrapper(props));
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('on user error show error container', () => {
      const wrapper = mount(authenticationWrapper(props));
      expect(wrapper.find(ErrorContainer).exists()).toBe(false);
      wrapper.setProps({ errors: { userError: 'user exists' }});
      expect(wrapper.find(ErrorContainer).exists()).toBe(true);
    });

    describe('inputs logic', () => {
      it('send request if inputs are not empty', () => {
        const wrapper = mount(authenticationWrapper(props));
        wrapper.find('div.button').simulate('click');
        expect(props.onUserRequest).toHaveBeenCalledTimes(0);
        wrapper.setState({ username: 'test', password: '123' });
        wrapper.find('div.button').simulate('click');
        expect(props.onUserRequest).toHaveBeenCalledTimes(1);
      });

      it('check inputs content with their sync states', () => {
        const wrapper = mount(authenticationWrapper(props));
        wrapper.setState({ username: 'test', password: '123' });
        expect(wrapper.find('input[type="text"]').props().value === 'test').toBe(true);
        expect(wrapper.find('input[type="password"]').props().value === '123').toBe(true);
      });
    });
  });
});
