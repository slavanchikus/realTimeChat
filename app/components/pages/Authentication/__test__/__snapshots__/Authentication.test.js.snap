// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication when requred props are defined for reply Authentication renders correctly 1`] = `
<Authentication
  errors={Object {}}
  onUserCreate={[MockFunction]}
  onUserRequest={[MockFunction]}
>
  <div
    className="container"
  >
    <h2>
      Авторизация
    </h2>
    <input
      maxLength={15}
      onChange={[Function]}
      placeholder="Юзернейм"
      type="text"
      value=""
    />
    <input
      maxLength={10}
      onChange={[Function]}
      placeholder="Пароль"
      type="password"
      value=""
    />
    <div
      className="button"
      onClick={[Function]}
    >
      Войти
    </div>
    <div
      className="choose"
      onClick={[Function]}
    >
      Зарегистрироваться
    </div>
  </div>
</Authentication>
`;
